### PR TITLE
Client expo auth session refactor and tests

### PR DESCRIPTION
**Pull request by prastoin**
close https://github.com/AdonisEnProvence/MusicRoom/issues/544

### CODE REVIEW

File: packages/client/__tests__/GoogleAuthentication.test.tsx [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+                        {
+                            target: 'Rendering signing in screen.Server user access token verification error.Unavailable related nickname or email',
+                        },
+                    'Make server send back an nickname nor email invalid error':
+                        {
+                            target: 'Rendering signing in screen.Server user access token verification error.Invalid related nickname or email',
+                        },
+                    'Make server send back an unknown error': {
+                        target: 'Rendering signing in screen.Server user access token verification error.Unknown server error',
+                    },
```
Comment (Devessier) [inline packages/client/__tests__/GoogleAuthentication.test.tsx:220]: **💭 thought:** ‏When dealing with events in `@xstate/test` that modify the setup, I prefer to dissociate them from actions that a user could actually do.

It means that I would not call `pressContinueWithGoogleButton()` in the same events that modify how the server responds. To keep tracks in the machine that the setup has changed, I would assign to the context a boolean indicating that, like here: https://github.com/AdonisEnProvence/MusicRoom/blob/f38450da3a4fa7924bbd3696ffec3709683963a7/packages/client/__tests__/AuthenticationPasswordReset.test.tsx#L39.

But, to be honest, there is a lot of possible cases. And writing all these booleans, and a guard for each, would be tedious!

**Review by Devessier — APPROVED** [review]
Really nice job!
I like the integration of Google OAuth with XState!
We can merge.

File: packages/client/tests/server/handlers.ts [hunk 2]
```diff
@@ -935,10 +945,71 @@ export const handlers = [
...
+                ctx.json({
+                    type: 'success',
+                    errorCode: null,
+                    authentication,
+                    params: {},
+                    url: '',
+                    error: undefined,
                 }),
             );
         },
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/client/tests/server/handlers.ts:1015]: Not me :eyes:

File: packages/client/__tests__/GoogleAuthentication.test.tsx (hunk 1 above)
Comment (prastoin) [reply packages/client/__tests__/GoogleAuthentication.test.tsx:220]: Indeed you're completely right !
As you say a lot to deal with I went to the fast solution

**Pull request merged**